homeassistant/components/plant.py

```diff
@@ -182,10 +191,11 @@ def state_changed(self, entity_id, _, new_state):
         value = new_state.state
         _LOGGER.debug("Received callback from %s with value %s",
                       entity_id, value)
-        if value == STATE_UNKNOWN:
+        if value == STATE_UNKNOWN or not value:
             return
 
         reading = self._sensormap[entity_id]
+        self._max_age_map[reading] = datetime.now()
```

Comment:
Rather than locally caching the "age", would it make sense to directly access the sensor's `last_updated` attribute when you loop through the sensors in the update() routine e.g. self.hass.states.get(sensor_entity_id)